### PR TITLE
Add controller code for rsync-tls mover

### PR DESCRIPTION
**Pull request by JohnStrunk**
**Describe what this PR does**
- [x] API types for rsync-tls mover
- [x] Operator code to manage mover
- [x] Updates to Helm chart to deploy the new mover
- [x] Updates to OLM manifests for new mover

**Is there anything that requires special attention?**

**Related issues:**
- Related to #364 
- Follow-on to #510 
- Requires openshift/release#34231
- Docs in #516

### CODE REVIEW

**Comment by JohnStrunk** [comment]
/cc @tesshuflower
I think this is ready... e2e finally passes on OCP.

File: controllers/mover/rsynctls/builder.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+		volumehandler.WithOwner(destination),
+		volumehandler.FromDestination(&destination.Spec.RsyncTLS.ReplicationDestinationVolumeOptions),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Mover{
+		client:               client,
+		logger:               logger.WithValues("method", "Rsync"),
```
Comment (tesshuflower) [inline controllers/mover/rsynctls/builder.go:157]: Should this be "RsyncTLS"?

Comment (JohnStrunk) [reply controllers/mover/rsynctls/builder.go]: Yes... fixed.

**Review by tesshuflower — APPROVED** [review]
/lgtm

Nice work John!

**Pull request merged**